TestsUpdateUsers.cs

```diff
@@ -0,0 +1,217 @@
+        private RestRequest _requestForWriteScope;
+
+        [SetUp]
+        public void Setup()
+        {
+            var clientForReadScope = ClientForReadScope.GetInstance();
+            _clientForReadScope = clientForReadScope.GetRestClient();
+            _requestForReadUsers = new RestRequest("/users");
+            _requestForReadZipCodes = new RestRequest("/zip-codes");
+
+            var clientForWriteScope = ClientForWriteScope.GetInstance();
+            _clientForWriteScope = clientForWriteScope.GetRestClient();
+            string requestMethodParam = TestContext.Parameters["requestMethodParamForUpdateUsers"];
+            Method requestMethod = Method.Put;
+            if (requestMethodParam == "Patch")
+            {
+                requestMethod = Method.Patch; 
+            }
+            _requestForWriteScope = new RestRequest("/users", requestMethod);
+        }
+
+        [Test]
+        public void UpdateAnyUserWithAllValidData_CheckTheUserWasUpdated_Test()
+        {
+            _clientForReadScope.AddDefaultHeader("Accept", "application/json");
+            RestResponse getAllUsers = _clientForReadScope.Execute(_requestForReadUsers);
+            List<User> allUsers = JsonConvert.DeserializeObject<List<User>>(getAllUsers.Content);
+            int initialCountUsers = allUsers.Count;
+            User selectedUser = allUsers[0];
+
+            RestResponse getZipCodesResponse = _clientForReadScope.Execute(_requestForReadZipCodes);
+            List<string> availableZipCodes = JsonConvert.DeserializeObject<List<string>>(getZipCodesResponse.Content);
+            string availableZipCode = availableZipCodes[0];
+
+            var updateUser = new
+            {
+                userNewValues = new
+                {
+                    age = RandomUserGenerator.GenerateRandomAge(),
+                    name = RandomUserGenerator.GenerateRandomName(),
+                    sex = RandomUserGenerator.GenerateRandomSex(),
+                    zipCode = availableZipCode
+
+                },
+                userToChange = new
+                {
+                    age = selectedUser.Age,
+                    name = selectedUser.Name,
+                    sex = selectedUser.Sex,
+                    zipCode = selectedUser.ZipCode
+                }
+            };
+            _requestForWriteScope.AddJsonBody(updateUser);
+            RestResponse updateResponse = _clientForWriteScope.Execute(_requestForWriteScope);
+
+            // check if user is updated
+            RestResponse getUsersResponse = _clientForReadScope.Execute(_requestForReadUsers);
+            List<User> updatedUsers = JsonConvert.DeserializeObject<List<User>>(getUsersResponse.Content);
+            bool updatedUserFound = updatedUsers.Any(u => u.Age == updateUser.userNewValues.age && u.Name == updateUser.userNewValues.name && u.Sex == updateUser.userNewValues.sex && u.ZipCode == updateUser.userNewValues.zipCode);
```

Comment:
Probably, if you will implement methods to compare objects of User type, in this case it would be enough to verify received data like this - bool updatedUserFound = updatedUsers.Any(updateUser);. Again, it's not necessary, but you can try it if there is such wish on your side:)

Reply:
updateUser hasn't type User, I guess it is not possible to compare objects of User type here